wallet/resync/resync_manager.go

```diff
@@ -30,42 +31,50 @@ func (r *ResyncManager) Start() {
 }
 
 func (r *ResyncManager) CheckUnfunded() {
-	unfunded, err := r.sales.GetNeedsResync()
+	unfundedSales, err := r.sales.GetUnfunded()
 	if err != nil {
 		log.Error(err)
 		return
 	}
-	if len(unfunded) == 0 {
+	unfundedPurchases, err := r.purchases.GetUnfunded()
+	if err != nil {
+		log.Error(err)
 		return
 	}
-	wallets := make(map[string]time.Time)
-	rollbackTime := time.Unix(2147483647, 0)
-	if r.mw != nil {
-		for cc := range r.mw {
-			wallets[strings.ToUpper(cc.CurrencyCode())] = rollbackTime
-		}
+	unfunded := append(unfundedSales, unfundedPurchases...)
+	if len(unfunded) == 0 {
+		return
 	}
+	wallets := make(map[string][]string)
 	for _, uf := range unfunded {
-		t, ok := wallets[strings.ToUpper(uf.PaymentCoin)]
+		addrs, ok := wallets[strings.ToUpper(uf.PaymentCoin)]
 		if !ok {
-			log.Warningf("ResyncManager: no wallet for sale with payment coin %s", uf.PaymentCoin)
-			continue
-		}
-		if uf.Timestamp.Before(t) {
-			t = uf.Timestamp.Add(-time.Hour * 24)
-			wallets[strings.ToUpper(uf.PaymentCoin)] = t
+			addrs = []string{}
+			wallets[strings.ToUpper(uf.PaymentCoin)] = addrs
 		}
-		r.sales.SetNeedsResync(uf.OrderId, false)
+		addrs = append(addrs, uf.PaymentAddress)
+		wallets[strings.ToUpper(uf.PaymentCoin)] = addrs
 	}
 	if r.mw != nil {
-		for cc, rbt := range wallets {
+		for cc, addrs := range wallets {
 			wal, err := r.mw.WalletForCurrencyCode(cc)
 			if err != nil {
 				log.Warningf("ResyncManager: no wallet for sale with payment coin %s", cc)
 				continue
 			}
-			log.Infof("Rolling back %s blockchain %s looking for payments for %d orders\n", cc, time.Since(rbt), len(unfunded))
-			wal.ReSyncBlockchain(rbt)
+			for _, addr := range addrs {
+				iaddr, err := wal.DecodeAddress(addr)
+				if err != nil {
+					log.Errorf("Error decoding unfunded payment address: %s", err)
+					continue
+				}
+				if err := wal.AddWatchedAddress(iaddr); err != nil {
+					log.Errorf("Error adding watched address: %s", err)
+				}
+			}
+
+			log.Infof("Rescanning blocking for %d orders\n", cc, len(unfunded))
```

Comment:
Can we make the error clearer than "blocking"? I'm not sure what you're trying to communicate with this error. (Also remove the trailing newline?)